src/subplots.ts

```diff
@@ -269,85 +265,118 @@ export class PlotPieChart extends PlotData {
         this.selected_areas = [];
         this.refresh_MinMax();
         this.plotObject = PieChart.deserialize(data);
-        this.plotObject.pieParts = this.plotObject.definePieParts([0, 0], this.height * 0.475);
+        this.plotObject.pieParts = this.plotObject.definePieParts(this.height * 0.475);
         this.color_to_plot_data = this.initHiddenColors()
     }
 
+  private initHiddenColors(): Map<string,PiePart> {
+    let hiddenColorsList: Map<string,PiePart> = new Map();
+    this.plotObject.pieParts.forEach(
+      part => {hiddenColorsList.set(part.hidden_color, part);})
+    return hiddenColorsList
+  }
+
   refresh_MinMax(): void {
     this.minX = -this.width / 2;
     this.maxX = this.width / 2;
     this.minY = -this.height / 2;
     this.maxY = this.height / 2;
   }
 
+  private drawPiechart(d: PieChart, mvx: number, mvy: number): void {
+    this.context_hidden.lineWidth = 0.5 / this.scaleX;
+    this.context_show.lineWidth = 0.5 / this.scaleY;
+    for (let part of d.pieParts) {
+      if (part.clicked) {
```

Comment:
As I said above, anything related to the mouseHandler will be the purpose of a future PR. 
Could you just add a comment that says that this should not be done in a draw method and that it will be moved later ?

Reply:
Done